### PR TITLE
refactor(fslib): prefer early return in resolveFilename

### PR DESCRIPTION
**Pull request by trivikr**
**What's the problem this PR addresses?**

Improves the readability of the code.

**How did you fix it?**

Used early return instead of break inside while loop.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-fslib/sources/ZipFS.ts [hunk 1]
```diff
@@ -815,21 +815,17 @@ export class ZipFS extends BasePortableFakeFS {
...
       if (!resolveLastComponent || this.symlinkCount === 0)
-        break;
+        return resolvedP;
 
       const index = this.libzip.name.locate(this.zip, resolvedP.slice(1));
       if (index === -1)
-        break;
+        return resolvedP;
+      if (!this.isSymbolicLink(index))
+        return resolvedP;
```
Comment (merceyz) [inline packages/yarnpkg-fslib/sources/ZipFS.ts:824]: You could combine these into a single if statement. I would make it a suggestion but GitHub wont let me.

Comment (trivikr) [reply packages/yarnpkg-fslib/sources/ZipFS.ts]: Done in [`d4657dc` (#4103)](https://github.com/yarnpkg/berry/pull/4103/commits/d4657dc4a94dfb1552ee4eb1b69c40f1997687e3)

**Comment by arcanis** [comment]
For note, I'm not particularly open to PRs that only make things "clearer" in subjective ways.

1. If everyone was doing it it'd be a pain
2. Clearer for someone isn't for everyone
3. It adds PRs to review, which push work on us
3. It can add bugs or regression (like your work with const)

While I appreciate your intent, helping the project and learning how it works would be better served by focusing on small practical tasks (we keep a list of [good first issues](https://github.com/yarnpkg/berry/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) to help with that).

**Comment by arcanis** [comment]
I prefer the current style on this PR

**Pull request closed without merging**